ui/src/components/RoleSelectScreen.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useState} from 'react'
+import { Button, Card } from 'semantic-ui-react'
+
+import { useParty, useStreamQuery, useLedger } from '@daml/react'
+import { UserSession } from '@daml.js/da-marketplace/lib/Marketplace/Role'
+
+import { OPERATOR_PARTY } from '../config'
+
+import TopMenu from './common/TopMenu'
+import OnboardingTile from './common/OnboardingTile'
+import { ArrowRightIcon } from '../icons/Icons'
+
+import { Mode } from './MainScreen'
+import './RoleSelectScreen.css'
+
+enum MarketRole {
+    INVESTOR = "Investor",
+    ISSUER = "Issuer",
+    CUSTODIAN = "Custodian"
+}
+
+type RoleSelectProps = {
+    loading: boolean;
+    caption: string;
+    disabled?: boolean;
+    roleSelectClick: () => void;
+}
+
+const RoleSelect: React.FC<RoleSelectProps> = ({ loading, disabled, caption, roleSelectClick }) => (
+    <Card className='centered role-select'>
+        <Button
+            disabled={disabled}
+            className='ghost'
+            loading={loading}
+            onClick={roleSelectClick}
+        >
+            { caption } <ArrowRightIcon/>
+        </Button>
+    </Card>
+)
+
+type Props = {
+    setView: (view: Mode) => void;
+    onLogout: () => void;
+}
+
+const RoleSelectScreen: React.FC<Props> = ({ setView, onLogout }) => {
+    const user = useParty();
+    const ledger = useLedger();
+    const userSessions = useStreamQuery(UserSession).contracts;
+
+    const [ loading, setLoading ] = useState(false);
+    const [ role, setRole ] = useState<MarketRole>();
+
+    const handleRoleClick = async (role: MarketRole, view: Mode) => {
+        setRole(role);
+        if (userSessions.length === 0) {
+            setLoading(true);
+            await ledger.create(UserSession, {
+                user,
+                role,
+                operator: OPERATOR_PARTY
```

Comment:
we would need to change this. (If in DABL we query for the .well-known parties and we use the UserAdmin as the operator party